go/engine/saltpack_user_keyfinder.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package engine
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/libkb"
+
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+// SaltpackUserKeyfinder is an engine to find Per User Keys (PUK). Users can also be loaded by assertions, possibly tracking them if necessary.
+// This engine does not find per team keys, which capability is implemented by SaltpackRecipientKeyfinder in the saltpackKeyHelpers package.
+type SaltpackUserKeyfinder struct {
+	Arg                           libkb.SaltpackRecipientKeyfinderArg
+	RecipientEntityKeyMap         map[keybase1.UserOrTeamID]([]keybase1.KID)
+	RecipientDeviceAndPaperKeyMap map[keybase1.UID]([]keybase1.KID)
+}
+
+// NewSaltpackUserKeyfinderAsInterface creates a SaltpackUserKeyfinder engine.
+func NewSaltpackUserKeyfinderAsInterface(Arg libkb.SaltpackRecipientKeyfinderArg) libkb.SaltpackRecipientKeyfinderEngineInterface {
```

Comment:
Usually it's better style to return `*SaltpackUserKeyfinder` and then have an assertion it's the right type.  This style of return might cause the classic `nil` interface bug.

Reply:
ok, fine to leave the signature, but can this constructor wrap the next constructor?